extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/utils/UiAssetMapper.java

```diff
@@ -140,19 +143,44 @@ public UiAsset buildUiAsset(JsonObject assetJsonLd, String connectorEndpoint, St
                 HttpDatasourceHints.BODY,
                 HttpDatasourceHints.METHOD,
                 HttpDatasourceHints.PATH,
-                HttpDatasourceHints.QUERY_PARAMS
+                HttpDatasourceHints.QUERY_PARAMS,
+
+                Prop.SovityDcatExt.CUSTOM_JSON
         ));
-        uiAsset.setAdditionalProperties(getStringProperties(remaining));
-        uiAsset.setAdditionalJsonProperties(getJsonProperties(remaining));
+
+        // custom properties
+        val serializedJsonLd = packRemainingJsonLdProperties(remaining);
+        uiAsset.setCustomJsonLdAsString(serializedJsonLd);
 
         // Private Properties
-        var privateProperties = JsonLdUtils.tryCompact(getPrivateProperties(assetJsonLd));
-        uiAsset.setPrivateProperties(getStringProperties(privateProperties));
-        uiAsset.setPrivateJsonProperties(getJsonProperties(privateProperties));
+        val privateProperties = JsonLdUtils.object(assetJsonLd, Prop.Edc.PRIVATE_PROPERTIES);
+        if (privateProperties != null) {
+            val privateCustomJson = JsonLdUtils.string(privateProperties, Prop.SovityDcatExt.PRIVATE_CUSTOM_JSON);
+            uiAsset.setPrivateCustomJsonAsString(privateCustomJson);
+
+            val privateSerializedJsonLd = packPrivateProperties(privateProperties);
+            uiAsset.setPrivateCustomJsonLdAsString(privateSerializedJsonLd);
+        }
 
         return uiAsset;
     }
 
+    private String packPrivateProperties(JsonObject privateProperties) {
+        val withoutCustomJson = Json.createObjectBuilder(privateProperties)
+                .remove(Prop.SovityDcatExt.PRIVATE_CUSTOM_JSON)
+                .build();
+        val compacted = JsonLdUtils.tryCompact(withoutCustomJson);
+        return JsonUtils.toJson(compacted);
+    }
+
+    private static String packRemainingJsonLdProperties(JsonObject remaining) {
```

Comment:
Maybe this should be called `packJsonLdProperties`. The context that they are remaining is independent of this function